react/Modal/index.jsx

```diff
@@ -9,9 +9,94 @@ import migrateProps from '../helpers/migrateProps'
 import palette from '../../stylus/settings/palette.json'
 import Portal from 'preact-portal'
 
-const ModalContent = ({ children, className }) => (
-  <div className={cx(styles['c-modal-content'], className)}>{children}</div>
-)
+class ModalContent extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { displayGhostHeader: false }
+  }
+
+  componentDidMount() {
+    this.scrollingContent.addEventListener('scroll', this.handleScroll, {
+      passive: true
+    })
+  }
+
+  handleScroll = () => {
+    if (
+      !this.state.displayGhostHeader &&
+      this.scrollingContent.scrollTop > 56
+    ) {
+      this.setState(() => ({ displayGhostHeader: true }))
+    } else if (
+      this.state.displayGhostHeader &&
+      this.scrollingContent.scrollTop < 56
+    ) {
+      this.setState(() => ({ displayGhostHeader: false }))
+    }
+  }
+
+  render() {
+    const { iconSrc, iconDest, service, children, className } = this.props
+
+    const { displayGhostHeader } = this.state
+
+    return (
+      <div
+        className={cx(styles['c-modal-content'], className)}
+        ref={div => {
+          this.scrollingContent = div
+        }}
+      >
+        {iconSrc &&
+          iconDest && (
+            <div>
+              <h2 className={cx(styles['c-modal-header--icon'], className)}>
+                <div>
+                  <Icon icon={iconSrc} className={styles['c-header-icon']} />
+                  {service && (
+                    <Icon
+                      icon="gear"
+                      color={palette['silver']}
+                      className={styles['c-header-service']}
+                    />
+                  )}
+                </div>
+                <Icon
+                  icon="exchange"
+                  className={styles['c-header-exchange']}
+                  color={palette['coolGrey']}
+                />
+                <Icon icon={iconDest} className={styles['c-header-icon']} />
+              </h2>
+              <div
```

Comment:
If anyone has a better idea to duplicate an element without explicitly duplicate it, I'm all ears. I couldn't find one but maybe someone has a better idea.
NB: I need the `h2` to be duplicated so I can use it without interfering with the flow.